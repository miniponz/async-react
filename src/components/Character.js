import React from 'react';
import PropTypes from 'prop-types';

function Character({ name, status, species, image }) {
  return (
    <section>
      <h3>{name}</h3>
      <img src={image}></img>
      <p>{status}</p>
      <p>{species}</p>
    </section>
  );
}

Character.propTypes = {
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired
};

export default Character;
