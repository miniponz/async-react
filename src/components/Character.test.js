import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('character component', () => {
  it('renders one lonely character', () => {
    const wrapper = shallow(<Character species ={''} name={''} status={''} image={''}/>);
    expect(wrapper).toMatchSnapshot();
  });   
});
