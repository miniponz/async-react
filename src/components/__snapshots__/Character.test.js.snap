// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character component renders one lonely character 1`] = `
<section>
  <h3 />
  <img
    src=""
  />
  <p />
  <p />
</section>
`;
